File: src/day1.js
console.log(require('fs').readFileSync("./inputs/day1.txt", {encoding: "utf8", flag: "r"}).trim().split("\n\n").map(i => i.split("\n").reduce((acc, a) => acc + parseInt(a), 0)).sort().reverse()[1])

console.log(require('fs').readFileSync("./inputs/day1.txt", {encoding: "utf8", flag: "r"}).trim().split("\n\n").map(i => i.split("\n").reduce((acc, a) => acc + parseInt(a), 0)).sort().reverse().slice(1,4).reduce((acc, x) => acc +x, 0))

File: 2022/src/day1.hs
import Data.List.Split as Split
import Data.List

main = do
  content <- readFile "../inputs/day1.txt"
  let input = lines content
  let result1 = maximum $ calories input
  let result2 = sum . take 3 . reverse . sort $ calories input
  putStrLn $ "part 1 : " ++ show result1
  putStrLn $ "part 2 : " ++ show result2

calories input = map (sum . map read) $ Split.splitOn [""] input

